proposals/nnnn-app-lib-vehicle-data-manager.md

```diff
@@ -0,0 +1,215 @@
+
+#### iOS
+
+##### Cached Data
+The iOS library will cache vehicle data in a similar way to how it caches system capabilities: each piece of vehicle data will be stored separately on the manager, like so:
+
+```objc
+@property (strong, nonatomic, nullable) SDLGPSData *gps;
+@property (nullable, strong, nonatomic) NSNumber<SDLFloat> *speed;
+@property (nullable, strong, nonatomic) NSNumber<SDLInt> *rpm;
+@property (nullable, strong, nonatomic) NSNumber<SDLFloat> *fuelLevel;
+@property (nullable, strong, nonatomic) SDLComponentVolumeStatus fuelLevelState;
+@property (nullable, strong, nonatomic) NSArray<SDLFuelRange *> *fuelRange;
+@property (nullable, strong, nonatomic) NSNumber<SDLFloat> *instantFuelConsumption;
+@property (nullable, strong, nonatomic) NSNumber<SDLFloat> *externalTemperature;
+@property (nullable, strong, nonatomic) NSString *vin;
+@property (nullable, strong, nonatomic) SDLPRNDL prndl;
+@property (nullable, strong, nonatomic) SDLTireStatus *tirePressure;
+@property (nullable, strong, nonatomic) NSNumber<SDLInt> *odometer;
+@property (nullable, strong, nonatomic) SDLBeltStatus *beltStatus;
+@property (nullable, strong, nonatomic) SDLBodyInformation *bodyInformation;
+@property (nullable, strong, nonatomic) SDLDeviceStatus *deviceStatus;
+@property (nullable, strong, nonatomic) SDLVehicleDataEventStatus driverBraking;
+@property (nullable, strong, nonatomic) SDLWiperStatus wiperStatus;
+@property (nullable, strong, nonatomic) SDLHeadLampStatus *headLampStatus;
+@property (nullable, strong, nonatomic) NSNumber<SDLFloat> *engineOilLife;
+@property (nullable, strong, nonatomic) NSNumber<SDLFloat> *engineTorque;
+@property (nullable, strong, nonatomic) NSNumber<SDLFloat> *accPedalPosition;
+@property (nullable, strong, nonatomic) NSNumber<SDLFloat> *steeringWheelAngle;
+@property (nullable, strong, nonatomic) SDLECallInfo *eCallInfo;
+@property (nullable, strong, nonatomic) SDLAirbagStatus *airbagStatus;
+@property (nullable, strong, nonatomic) SDLEmergencyEvent *emergencyEvent;
+@property (nullable, strong, nonatomic) SDLClusterModeStatus *clusterModeStatus;
+@property (nullable, strong, nonatomic) SDLMyKey *myKey;
+@property (nullable, strong, nonatomic) SDLElectronicParkBrakeStatus electronicParkBrakeStatus;
+@property (nullable, strong, nonatomic) SDLTurnSignal turnSignal;
+@property (nullable, strong, nonatomic) NSString *cloudAppVehicleID;
+```
+
+Note that the above is a 1-1 representation of `OnVehicleData`, however, this may not always be the case. For example, with the change recommended in [SDL-0256 Refactor Fuel Information Related Data](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0256-Refactor-Fuel-Information-Related-Vehicle-Data.md), the maintainers may decide that it is best to deprecate all the old elements and to coerce the old data types into the new data structure automatically.
```

Comment:
```suggestion
Note that the above is a 1-1 representation of `OnVehicleData`, however, this may not always be the case. For example, with the change recommended in [SDL-0256 Refactor Fuel Information Related Data](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0256-Refactor-Fuel-Information-Related-Vehicle-Data.md), the SDLC Project Maintainer may decide that it is best to deprecate all the old elements and to coerce the old data types into the new data structure automatically.
```